upgrade/util/util.go

```diff
@@ -144,3 +148,770 @@ func CreateOrUpdateLonghornVersionSetting(namespace string, lhClient *lhclientse
 	}
 	return nil
 }
+
+func ListSettings(namespace string, lhClient *lhclientset.Clientset, resourceMaps map[string]interface{}) (map[string]*v1beta2.Setting, error) {
+	if v, ok := resourceMaps[types.LonghornKindSetting]; ok {
+		return v.(map[string]*v1beta2.Setting), nil
+	}
+
+	settings := map[string]*v1beta2.Setting{}
+	settingList, err := lhClient.LonghornV1beta2().Settings(namespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	for _, setting := range settingList.Items {
+		settingCopy := v1beta2.Setting{}
+		if err := copier.Copy(&settingCopy, setting); err != nil {
+			return nil, err
+		}
```

Comment:
Any reason to copy objects?

Reply:
Do you mean directly assigne `setting`'s address to `settings[name]`?

Reply:
Yes. Is there any concern missed?

Reply:
Yeah. No need to have an extra copy because the result is from the API client directly instead of cache.

Reply:
Originaly, I used the method I mentioned in https://github.com/longhorn/longhorn-manager/pull/1698#discussion_r1103978644 but met an issue, so I use `copy` instead.

Now, I found the root cause https://husni.dev/beware-of-implicit-memory-aliasing-in-go-foor-loop/.

Reply:
I see what you encountered. And I had handled some similar cases somewhere. Hence I would recommend the format I used:
```
for i := range settingList.Items {
    setting := settingList.Items[i]
    ......
}
```

Reply:
Addressed. Thank you.